src/index.ts

```diff
@@ -0,0 +1,108 @@
+import * as core from "@actions/core";
+import { File as DiffFile } from "parse-diff";
+import minimatch from "minimatch";
+import { callModel, createReviewPrompt, parseAIResponse } from "./ai";
+import { ReviewComment } from "./types";
+import {
+  getFileContent,
+  getPRContext,
+  getRepoContext,
+  submitReviewComments,
+} from "./git";
+
+// TODO: add conditional review when added as reviewer
+// TODO: customize the bot name and avatar
+// TODO: set up cicd to deploy the action
+
+async function prepareReviewContext(
+  prContext: Awaited<ReturnType<typeof getPRContext>>
+) {
+  const { pr, files, parsedDiff, owner, repo } = prContext;
+
+  const repoContext = await getRepoContext(owner, repo);
+  const excludePatterns = core
+    .getInput("exclude")
+    .split(",")
+    .map((s) => s.trim());
+  const filesWithContent = await Promise.all(
+    parsedDiff.map(async (file: DiffFile) => {
+      const filePath = file.to;
+      if (
+        !filePath ||
+        file.to === "/dev/null" ||
+        excludePatterns.some((pattern) => minimatch(filePath, pattern))
+      ) {
+        return null;
+      }
+
+      const content = await getFileContent(owner, repo, filePath, pr.head.sha);
+      return content ? { file, content } : null;
+    })
+  ).then((results) =>
+    results.filter(
+      (result): result is NonNullable<typeof result> => result !== null
+    )
+  );
+
+  return {
+    prContext: {
+      title: pr.title,
+      description: pr.body || "",
+      changedFiles: files.map((f) => f.filename),
+    },
+    repoContext,
+    filesContext: filesWithContent,
+  };
+}
+
+async function performAIReview(
+  reviewContext: Awaited<ReturnType<typeof prepareReviewContext>>
```

Comment:
Similar to the `prepareReviewContext` function, consider using a more specific type for `reviewContext` instead of `Awaited<ReturnType<typeof prepareReviewContext>>` to improve readability and maintainability.